internal/webhook/template_webhook.go

```diff
@@ -119,7 +120,39 @@ func (*ServiceTemplateValidator) ValidateUpdate(_ context.Context, _ runtime.Obj
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ServiceTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (in *ServiceTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	tmpl, ok := obj.(*v1alpha1.ServiceTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ServiceTemplate but got a %T", obj))
+	}
+
+	var token string
+
+	for {
+		managedClusters := &v1alpha1.ManagedClusterList{}
+		if err := in.Client.List(ctx, managedClusters, &client.ListOptions{
+			Namespace: tmpl.Namespace,
+			Limit:     100,
+			Raw: &metav1.ListOptions{
+				Continue: token,
+			},
+		}); err != nil {
+			return nil, err
+		}
+
+		for _, m := range managedClusters.Items {
+			for _, s := range m.Spec.Services {
+				if s.Template == tmpl.Name {
```

Comment:
could be interchanged with a single call to `slices.Contains*`, probably it'd be easier to read